### PR TITLE
Make a start on the grafter^W trace builder.

### PR DESCRIPTION
**Pull request by vext01**
First commit is the bulk of the work. This is just boilerplate at this point. Obviously it doesn't work yet.

Follow up commits are tidy ups.

### CODE REVIEW

File: ykrt/src/compile/jitc_yk/graft.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+//! The grafter.
+//!
+//! Assembles an in memory trace of Yk IR, by copying portions of the AOT IR.
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/graft.rs:3]: Oops, `in-memory`.

**Comment by vext01** [comment]
[dcefd3c](https://github.com/ykjit/yk/pull/906/commits/dcefd3ca5cc260c9e753fe047e7e29d419a3e15b) also fixes a copy-pasta.

**Comment by ptersilie** [comment]
I'm afraid I'm not sold on the name "grafter" yet.

**Comment by vext01** [comment]
Propose another.

I just don't want to have to say "jitmodbuilder" any more.

**Comment by vext01** [comment]
That last commit removes all references to "grafter"

File: ykrt/src/compile/jitc_yk/trace_builder.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+//! The trace builder.
+//!
+//! Given a mapped trace and an AOT module, assembles an in-memory Yk IR trace by copying blocks
+//! from the AOT IR. The output of this process will be the input to the code generator.
+
+use super::aot_ir::{self, Module};
+use crate::trace::{MappedTrace, TracedAOTBlock};
+use std::error::Error;
+
+struct Grafter<'a> {
```
Comment (ltratt) [inline ykrt/src/compile/jitc_yk/trace_builder.rs:10]: I guess this should be `TraceBuilder` now?

Comment (vext01) [reply ykrt/src/compile/jitc_yk/trace_builder.rs:10]: Huh! My grep-fu fails me!

**Comment by ltratt** [comment]
That one minor comment aside, this is OK with me.

**Comment by ptersilie** [comment]
Looks good to me aside from Laurie's comment.

**Comment by vext01** [comment]
OK to squash?

**Comment by vext01** [comment]
@ltratt are we good to squash here?

**Comment by ltratt** [comment]
Please squash.

**Comment by ltratt** [comment]
[Sorry, I wasn't assigned this one, so I didn't notice!]

**Comment by vext01** [comment]
That was unexpectedly annoying to squash!

But squashed anyway.

**Pull request merged**